tests/backup/backup_helper.go

```diff
@@ -1181,13 +1247,128 @@ func backupSuccessCheck(backupName string, orgID string, retryDuration int, retr
+	log.InfoD("Validation: Getting the backup objects (specs) from the Context")
+
+	var backupAppContexts []*scheduler.Context = make([]*scheduler.Context, 0)
+	var allBkpObjs []*api.ResourceInfo = make([]*api.ResourceInfo, 0)
+
+	//for each clusterAppsContext (namespace), we return the corresponding BackupSpecObject
+	for _, clusterAppsContext := range clusterAppsContexts {
+		var specObjects []interface{} = make([]interface{}, 0)
+		// the Backup objects which were found in this namespace
+		var resInfoObjs []*api.ResourceInfo = make([]*api.ResourceInfo, 0)
+
+		for _, resource := range resourceInfos {
+			if resource.Namespace == clusterAppsContext.ScheduleOptions.Namespace {
+				resInfoObjs = append(resInfoObjs, resource)
+			} else if resource.Namespace == "" {
+				// IF it is therein this context, we'll work with it
+				resInfoObjs = append(resInfoObjs, resource)
+			}
+		}
+
+	specloop:
+		for _, spec := range clusterAppsContext.App.SpecList {
+			name, kind, ns, err := GetSpecNameKindNamepace(spec)
+			if err != nil {
+				log.Error(err)
+				continue specloop
+			}
+			if name != "" /*&& kind != ""*/ /*kind is empty for certain objects :(*/ {
+				for _, bkpObj := range resInfoObjs {
+					if name == bkpObj.Name /*&& kind == bkpObj.Kind*/ {
+						clone := spec
+						// The spec for the object/resource was found in the Context (namespace). Appending it.
+						if ns == "" {
+							log.InfoD("The resource(name: %s, kind: %s, with NO NAMESPACE value) was found in the context '%s'", name, kind, clusterAppsContext.ScheduleOptions.Namespace)
+						}
+						specObjects = append(specObjects, clone)
+						continue specloop
+					}
+				}
+			} else {
+				log.Warnf("some error with getting GetSpecNameKindNamepace with Spec Name: %s, Kind: %s, Namespace: %s, in local Context (NS): %s", name, kind, ns, clusterAppsContext.ScheduleOptions.Namespace)
+				continue specloop
+			}
+			// executes if specObj is not in backup
+			log.Warnf("This Backup does not include the resource(name: %s, kind:%s, namespace:%s) found in the local context (NS) '%s'", name, kind, ns, clusterAppsContext.ScheduleOptions.Namespace)
+		}
+
+		// Duplicate the object
+		backupAppContext := *clusterAppsContext
+		// Duplicate the object
+		app := *clusterAppsContext.App
+		app.SpecList = specObjects
+		backupAppContext.App = &app
+		backupAppContexts = append(backupAppContexts, &backupAppContext)
+		allBkpObjs = append(allBkpObjs, resInfoObjs...)
+	}
+
+	// Error Checking: if the contexts for all backed up resources were found in SOME namespace
+
+	var namespaces []string = nil
+	getNS := func() {
+		if namespaces == nil {
+			for _, clusterAppsContext := range clusterAppsContexts {
+				namespaces = append(namespaces, clusterAppsContext.ScheduleOptions.Namespace)
+			}
+		}
+	}
+
+	for _, resInfo := range resourceInfos {
+		found := false
+		for _, bkpObj := range allBkpObjs {
```

Comment:
Do not use `bkp`, use `backup` as much as possible

Reply:
I have significantly shortened the error messages and even cleaned and consolidated logging, but I still left in _really short and crisp help messages_. After going through them, if you still feel that it is way too much OR unnecessary OR confusing in nature OR some other reason, I'll restrict it to error-only messages. 😄

full reply: https://github.com/portworx/torpedo/pull/1182#discussion_r1134952153

Reply:
✅